src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs

```diff
@@ -0,0 +1,223 @@
+    public const string CancelPendingChangeViewName = "~/Views/ChangeOfPrice/Employer/CancelPendingChange.cshtml";
+    public const string ApproveProviderChangeOfPriceViewName = "~/Views/ChangeOfPrice/Employer/ApproveProviderChangeOfPrice.cshtml";
+
+
+    public ChangeOfPriceEmployerController(
+        ILogger<ChangeOfPriceEmployerController> logger,
+        IApprenticeshipService apprenticeshipService,
+        IMapper mapper,
+        ICacheService cache,
+        UrlBuilder externalEmployerUrlHelper)
+    {
+        _logger = logger;
+        _apprenticeshipService = apprenticeshipService;
+        _mapper = mapper;
+        _cache = cache;
+        _externalEmployerUrlHelper = externalEmployerUrlHelper;
+    }
+
+    [HttpGet]
+    [SetNavigationSection(NavigationSection.ManageApprentices)]
+    [Route("employer/{employerAccountId}/ChangeOfPrice/{apprenticeshipHashedId}")]
+    public async Task<IActionResult> GetEmployerEnterChangeDetails(string apprenticeshipHashedId)
+    {
+        var apprenticeshipPrice = await GetApprenticeshipPrice(apprenticeshipHashedId);
+        if (apprenticeshipPrice == null)
+        {
+            return NotFound();
+        }
+
+        var model = _mapper.Map<EmployerChangeOfPriceModel>(apprenticeshipPrice);
+        PopulateEmployerInitiatedRouteValues(model);
+        await _cache.SetCacheModelAsync(model);
+        return View(EnterChangeDetailsViewName, model);
+    }
+
+    [HttpGet]
+    [Route("employer/{employerAccountId}/ChangeOfPrice/{apprenticeshipHashedId}/edit")]
+    public IActionResult GetEmployerEditChangeDetails(EmployerChangeOfPriceModel model)
+    {
+        return View(EnterChangeDetailsViewName, model);
+    }
+
+    [HttpPost]
+    [SetNavigationSection(NavigationSection.ManageApprentices)]
+    [Route("employer/{employerAccountId}/ChangeOfPrice/{apprenticeshipHashedId}")]
+    public async Task<IActionResult> EmployerCheckDetails(EmployerChangeOfPriceModel model)
+    {
+        if (!ModelState.IsValid)
+        {
+            return View(EnterChangeDetailsViewName, model);
+        }
+
+        await _cache.SetCacheModelAsync(model);
+        return View(CheckDetailsViewName, model);
+    }
+
+    [HttpPost]
+    [Route("employer/{employerAccountId}/ChangeOfPrice/{apprenticeshipHashedId}/submit")]
+    public async Task<IActionResult> EmployerInitiatedSubmitChange(EmployerChangeOfPriceModel model)
+    {
+        await _apprenticeshipService.CreatePriceHistory(model.ApprenticeshipKey, "Employer", HttpContext.User.GetUserId(), null, null, model.ApprenticeshipTotalPrice, HttpUtility.HtmlEncode(model.ReasonForChangeOfPrice), model.EffectiveFromDate.Date.GetValueOrDefault());
+
+        var employerCommitmentsReturnUrl = $"{_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", model.EmployerAccountId, model.ApprenticeshipHashedId.ToUpper())}?showChangeOfPriceRequestSent=true";
```

Comment:
I know this file was moved not modified, but there is a code warning on this line that would be worth resolving